frontend/src/components/doctor/EditDoctorInfo.tsx

```diff
@@ -1,40 +1,48 @@
 import DoctorForm from "./DoctorForm";
 import { useNavigate, useParams } from "react-router-dom";
-import {
-  fetchIndividualDoctorInfo,
-  saveIndividualDoctor,
-} from "../api/FetchData";
+
 import { Heading } from "@chakra-ui/react";
 import { useEffect, useState } from "react";
-import { DoctorDetail } from "./DetailDoctorPage";
+import { Doctor } from "./DoctorControl"; // type import
 
 function EditDoctorInfo() {
+  let [individualDoctor, setIndividualDoctor] = useState<Doctor>();
+
   const params = useParams();
   const navigate = useNavigate();
 
-  if (!params.id) {
-    return <div>No Doctor found</div>;
-  }
+  // ! means I know doctor id always exist -- don't give error
+  const doctorID = parseInt(params.id!, 10);
+
+  // get request
 
-  const doctorID = parseInt(params.id, 10);
+  useEffect(() => {
+    const getIndividualDoctor = async () => {
+      const request = await fetch(`/api/doctor/${doctorID}`);
+      const result = (await request.json()) as Doctor;
 
-  const individualDoctorDetail = fetchIndividualDoctorInfo(doctorID);
+      console.log(result);
+      setIndividualDoctor(result);
+    };
 
-  if (!individualDoctorDetail) {
-    return <div>Doctor not found</div>;
-  }
+    getIndividualDoctor();
+  }, []);
 
   return (
     <>
       <Heading>Update Doctor Info</Heading>
       <DoctorForm
-        individualDoctorDetail={individualDoctorDetail}
-        onSave={(details: DoctorDetail) => {
-          saveIndividualDoctor(details);
-          alert(
-            `You have successfully edited the ${details.first_name} information`
-          );
-          navigate("/");
+        individualDoctorDetail={individualDoctor}
+        onSave={async (doc: Doctor) => {
+          await fetch(`/api/doctor/${doctorID}`, {
+            method: "PUT",
+            body: JSON.stringify(doc),
+            headers: {
+              Accept: "application/json",
+              "Content-Type": "application/json",
+            },
+          });
```

Comment:
This is ok for now, but I'd move all the data fetching functions to a utility file in the future to help keep all the related code together. This is especially useful if you want to share common headers between all your fetch requests.